src/libcore/iter.rs

```diff
@@ -2205,6 +2297,35 @@ impl<I> Iterator for Fuse<I> where I: Iterator {
     }
 
     #[inline]
+    fn nth(&mut self, n: usize) -> Option<I::Item> {
+        if self.done {
+            None
+        } else {
+            let nth = self.iter.nth(n);
```

Comment:
I'm assuming the implementer of `nth` is obeying the spec. That is, `nth(n)` is equivalent to `n+1` calls of `next()`. And shouldn't that be *at most* `n+1` calls to `next`? `nth` always stops after the first `None